woocommerce/woocommerce
# Add the new `/orders/{id}/actions` endpoint

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR introduces the new `POST /orders/{id}/actions` endpoint in the `OrderActionsRestController` class.
It allows re-sending details about an order to the customer associated with it.

Closes https://github.com/woocommerce/woocommerce/issues/51797



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create an order if you don't have one in your store.
2. Go to `wp-admin/admin.php?page=wc-settings&tab=advanced&section=keys` and add a new key with `Read/Write` permissions.
3. Make the following request using the consumer key and secret generated in the previous step, to send the order details with the custom `email` param:
```bash
curl --insecure -u [KEY]:[SECRET] --request POST --url 'https://store.local/wp-json/wc/v3/orders/[ORDER_NUMBER]/actions/send_order_details' --header 'Content-Type: application/json'
```
4. It should respond with:
```
{"message": "Order details email sent to customer."}
```
5. Make sure you got a message in the specified email with the subject: `Details for order #{id} on {store}`.
6. 



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Add the new `POST /orders/{id}/actions` endpoint to allow re-sending the order details to customers.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
